internal/clients/init.go

```diff
@@ -198,3 +194,44 @@ func initializeClients(dic *di.Container) {
 		},
 	})
 }
+
+func initMessagingClient(ctx context.Context, wg *sync.WaitGroup, dic *di.Container) bool {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	configuration := container.ConfigurationFrom(dic.Get)
+
+	msgClient, err := messaging.NewMessageClient(
+		types.MessageBusConfig{
+			PublishHost: types.HostInfo{
+				Host:     configuration.MessageQueue.Host,
+				Port:     configuration.MessageQueue.Port,
+				Protocol: configuration.MessageQueue.Protocol,
+			},
+			Type:     configuration.MessageQueue.Type,
+			Optional: configuration.MessageQueue.Optional,
+		})
+	if err != nil {
+		lc.Errorf("failed to create MessageClient: %v", err)
+		return false
+	}
+
+	err = msgClient.Connect()
+	if err != nil {
+		lc.Errorf("failed to connect to MessageClient: %v", err)
+		return false
+	}
+	go func() {
+		wg.Add(1)
```

Comment:
This call to `wg.Add` should be before starting the `go func`. I.e. just before 222.